File: JsonParseTest/results/JsonParseTest.pl
use strict;
use warnings;

my $obj1 = OneJson::parse("{ \"a\":1, \"b\":2 }");
if (!$obj1->is_object()) {
    die "expected to be object!"."\n";
}
my $obj1_props = $obj1->as_object()->get_properties();
if (scalar(@{$obj1_props}) != 2) {
    die "expected 2 properties"."\n";
}
if (${$obj1_props}[0]->get_name() ne "a") {
    die "expected first property to be named 'a'"."\n";
}
my $obj1_prop0_value = ${$obj1_props}[0]->get_value($obj1);
if (!$obj1_prop0_value->is_number() || $obj1_prop0_value->as_number() != 1) {
    die "expected 'a' to be 1 (number)"."\n";
}
print(("b = @{[$obj1->as_object()->get(\"b\")->as_number()]}") . "\n");

File: JsonParseTest/results/JsonParseTest.cpp
#include <one.hpp>
#include <iostream>
#include <stdexcept>
#include <vector>

int main()
{
    auto obj1 = OneJson::parse(std::string("{ \"a\":1, \"b\":2 }"));
    if (!obj1->isObject()) {
        throw std::runtime_error(std::string("expected to be object!"));
    }
    auto obj1_props = obj1->asObject()->getProperties();
    if (obj1_props->size() != 2) {
        throw std::runtime_error(std::string("expected 2 properties"));
    }
    if (obj1_props->at(0)->getName() != std::string("a")) {
        throw std::runtime_error(std::string("expected first property to be named 'a'"));
    }
    auto obj1_prop0_value = obj1_props->at(0)->getValue(obj1);
    if (!obj1_prop0_value->isNumber() || obj1_prop0_value->asNumber() != 1) {
        throw std::runtime_error(std::string("expected 'a' to be 1 (number)"));
    }
    std::cout << std::string("b = ") + std::to_string(obj1->asObject()->get(std::string("b"))->asNumber()) << std::endl;
    return 0;
}